src/plugins/intel_gpu/docs/dynamic_shape/kv_cache.md

```diff
@@ -0,0 +1,117 @@
+# KV cache management in stateful model
+## Description
+
+For auto-regressive transformer models, KV (key-value) cache plays a pivotal role to  used avoid duplicated computation for past tokens. In OpenVINO, management of the KV (key-value) cache differs between the [continuous batch (CB) pipeline](https://docs.openvino.ai/2025/model-server/ovms_demos_continuous_batching.html) and the non-CB pipeline. 
+
+In CB pipeline with openvino_genai, an SDPA (Scaled Dot Product Attention) operation in the OV IR (intermediate representation) is converted to  a PagedAttention opertation. Then, the KV cache memories are managed by CB pipeline. CB pipeline is the default mode of openvino_genai for running LLMs.
+
+On the other hand, we can use stateful model for Non-CB pipeline for KV cache optimization. In this case, KV cache memory is managed by the plugin. 
+
+In this document, KV cache management in stateful model within a non-CB pipeline is explained.
+
+
+## Basic structure of SDPA and kv cache in a stateful model
+
+The following diagram shows a typical pattern of the KV cache in a stateful LLM model.
+
+```dot
+digraph KVCacheFlow {
+    rankdir=TB;
+    node [shape=ellipse, fontsize=10];
+
+    StateInit     [label="State\nInitialize\nSubgraph"];
+    ReadValue     [label="ReadValue\n(Past KV)"];
+    BeamIdx       [label="Parameter\nbeam_idx"];
+    Gather        [label="Gather"];
+    KVProj        [label="KV projection\nfor new token"];
+    Concat        [label="Concat"];
+    SDPA          [label="SDPA"];
+    Assign        [label="Assign\n(Present KV)"];
+    VariableState      [shape=box, label="VariableState"];
+
+    // main flows
+    StateInit -> ReadValue;
+    ReadValue -> VariableState [style=dashed];
+    ReadValue -> Gather;
+    BeamIdx -> Gather;
+    Gather -> Concat;
+    KVProj -> Concat;
+    Concat -> SDPA;
+    Concat -> Assign;
+    Assign -> VariableState [style=dashed];
+}
```

Comment:
nit: this is not rendered in github.

Link to this MD file(graph is not rendered): https://github.com/yeonbok/openvino/blob/taylor_doc_update/src/plugins/intel_gpu/docs/dynamic_shape/kv_cache.md

Reply:
fixed